music_detection/staff.py

```diff
@@ -1,13 +1,62 @@
+from .measure import Measure
 from .key_enum import KeyEnum
 
+import cv2
+import numpy as np
 
 class Staff:
-    def __init__(self):
+    def __init__(self, image=None):
         self.key = KeyEnum.UNDEFINED
         self.time_signature = 0
-        self.position_in_image = []
         self.measure_list = []
+        self.image = image #Actual picture of the staff
+        self.line_gap = 0 #Added it just to signify that i need it from a previous processing
+
+    def identify_measures(self) -> None:
+        '''Split the staff into different Measure objects stored in measure_list'''
+        vote_threshold = int(3.25*self.line_gap) #Determined experimentally
+
+        #Preprocessing : will be probably be removed in later versions, because
+        #performed in another function
+        im_gray = cv2.cvtColor(self.image, cv2.COLOR_BGR2GRAY)
+        im_blur = cv2.GaussianBlur(im_gray, (3, 3), 0)
+        im_adapt_thresh = cv2.adaptiveThreshold(im_blur, 255, cv2.ADAPTIVE_THRESH_MEAN_C,
+                            cv2.THRESH_BINARY_INV, 11, 2)
+        #Closing operation
+        kernel = cv2.getStructuringElement(cv2.MORPH_RECT, (3, 3))
+        im_morph = cv2.morphologyEx(im_adapt_thresh, cv2.MORPH_CLOSE, kernel)
+
+        #Skeletonization
+        im_skel = np.zeros(im_morph.shape, np.uint8)
+        element = cv2.getStructuringElement(cv2.MORPH_CROSS, (3, 3))
+        while cv2.countNonZero(im_morph) != 0:
+            im_open = cv2.morphologyEx(im_morph, cv2.MORPH_OPEN, element)
+            temp = cv2.subtract(im_morph, im_open)
+            eroded = cv2.erode(im_morph, element)
+            im_skel = cv2.bitwise_or(im_skel, temp)
+            im_morph = eroded.copy()
+
+        #Extraction of bar lines'x-coordinates
+        lines = cv2.HoughLines(im_skel, 1, np.pi/180, vote_threshold)
+        bar_lines_loc = list()
+        for rho, theta in lines[:,0]:
+            if theta==0: #only vertical lines are kept
```

Comment:
Maybe the condition is a bit strict, consider taking a small threshold (like 5-10 degrees)

Reply:
Especially considering that the image rectification may not be perfect

Reply:
I know it is a bit strict, but I skeletize the image beforehand to make sure that I only keep vertical lines, thus avoiding such a case. I will look into it anyway and see what I can do.